File: pi/main/models/schedule.py
from .model import Model


class Schedule(Model):
    id = None
    day = None
    time = None
    amount = None
    groupId = None
    petId = None
    synced = None
    uploaded = None
    deleted = None
    uId = None
    tableName = 'schedule'
    serverId = None
    columnNames = [
        'id',
        'day',
        'feedTime',
        'amount',
        'groupId',
        'petId',
        'synced',
        'uploaded',
        'deleted',
        'uId',
        'serverId'
    ]

    def to_map(self):
        return {
            'id': self.id,
            'day': self.day,
            'feedTime': self.time,
            'amount': self.amount,
            'synced': self.synced,
            'groupId': self.groupId,
            'petId': self.petId,
            'uploaded': self.uploaded,
            'deleted': self.deleted,
            'uId': self.uId,
            'serverId': self.serverId
        }

    def from_map(self, data_map):
        if data_map is not None:

            if 'id' not in data_map:
                self.id = None
            else:
                self.id = data_map['id']

            self.day = data_map['day']
            self.time = data_map['feedTime']
            self.amount = data_map['amount']
            self.synced = 1 if data_map['synced'] else 0
            self.groupId = data_map['groupId']
            self.petId = data_map['petId']
            self.uploaded = 1 if data_map['uploaded'] else 0
            self.deleted = 1 if data_map['deleted'] else 0
            self.uId = data_map['uId']
            self.serverId = data_map['serverId']

File: petfeed/lib/src/data/database/models/schedule.dart
import 'package:petfeed/src/data/database/schedules/schedules_wirename.dart';

class Schedule {
  int id;
  double amount;
  String day;
  DateTime feedTime;
  String groupID;
  String uID;
  int petID;

  Map<String, dynamic> toMap() {
    Map<String, dynamic> map = <String, dynamic>{
      columnAmount: amount,
      columnDay: day,
      columnFeedTime: feedTime.millisecondsSinceEpoch,
      columnGroupId: groupID,
      columnUID: uID,
      columnPetId: petID,
    };

    if (id != null) {
      map[columnId] = id;
    }

    return map;
  }

  Schedule();

  Schedule.fromMap(Map<String, dynamic> map) {
    id = map[columnId];
    amount = map[columnAmount];
    day = map[columnDay];
    feedTime = DateTime.fromMillisecondsSinceEpoch(map[columnFeedTime].round());
    groupID = map[columnGroupId];
    uID = map[columnUID];
    petID = map[columnPetId];
  }
}
